src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -40,6 +40,24 @@ internal class AzureAppConfigurationProvider : ConfigurationProvider, IConfigura
         private static readonly TimeSpan MinDelayForUnhandledFailure = TimeSpan.FromSeconds(5);
         private static readonly TimeSpan DefaultMaxSetDirtyDelay = TimeSpan.FromSeconds(30);
 
+        public Uri AppConfigurationEndpoint
+        {
+            get
+            {
+                if (_options.Endpoint != null)
+                {
+                    return _options.Endpoint;
+                }
+
+                if (_options.ConnectionString != null)
+                {
+                    return new Uri(ConnectionStringParser.Parse(_options.ConnectionString, "Endpoint"));

```

Comment:
Where is `ConnectionStringParser` from?

Reply:
It's our own implementation at `\src\Microsoft.Extensions.Configuration.AzureAppConfiguration\ConnectionStringParser.cs` that is used in our offline cache scenario.

Reply:
I thought we didn't have a parse. Good :)